carapp/endpoints.py

```diff
@@ -79,15 +150,66 @@ def get(self):
         logger.info(f"\nList of Users: {list_of_users}")
         return {"Users": list_of_users}
 
-    @use_kwargs(user_args)
+
+class CarData(Resource):
+    def get(self, id):
+        car = Car.query.get(id)
+        if car is None:  # Comparision id of objects
+            return {"Reason": "Car not found", "Car_id": id}, 404
+        result = car_schema.dump(car)
+        logger.info(f"\nCar with id={id} was called")
+        return result
+
+    def delete(self, id):
+        car = Car.query.get(id)
+        result = car_schema.dump(car)
+        try:
+            result["id"]
+        except KeyError:
+            logger.error(f"\nNo Car with id={id}")
+            return {"Reason": f"No Car with id={id}"}, 404
+        db.session.delete(car)
+        db.session.commit()
+        logger.info(f"\nCar with id={id} deleted")
+        return "", 204
+
+    @use_kwargs(car_args)
+    def put(self, id, **kwargs):
+        # 0 or 1 (if updated)
+        car_updated = Car.query.filter_by(id=id).update(kwargs)
+        if not car_updated:
+            logger.error(f"\nNo Car with id={id}")
+            return {"Reason": "No such Car", "Car_id": id}, 404
+        result = car_schema.load({**{"id": id}, **kwargs})
+        db.session.commit()
+        logger.info(f"\nCar with id={id} updated")
+        return result, 200
+
+    @use_kwargs(car_args)
     def post(self, **kwargs):
-        user = User(**kwargs)
-        result = schema.dump(user)
-        db.session.add(user)
+        car = Car(**kwargs)
+        db.session.add(car)
         db.session.commit()
-        logger.info(f"\nUser with id={result['id']} created")
+        result = car_schema.dump(car)
+        logger.info(f"\nCar with id={result['id']} added")
         return result, 201
 
 
-api.add_resource(UserData, "/users/<int:user_id>")
-api.add_resource(UserList, "/users")
+class CarList(Resource):
+    def get(self):
+        kwargs = request.args  # get all parameters
+        car_list = Car.query.filter_by(**kwargs).all()
```

Comment:
I'm not sure about it. But probably if the field is `None` it will get entries with the field empty. Try it, please. If I'm right, do the following on the `kwargs` first:

`{k: v for k, v in kwargs.items() if v}`